source/server/listener_manager_impl.cc

```diff
@@ -335,6 +335,15 @@ ListenerManagerStats ListenerManagerImpl::generateStats(Stats::Scope& scope) {
 bool ListenerManagerImpl::addOrUpdateListener(const envoy::config::listener::v3::Listener& config,
                                               const std::string& version_info, bool added_via_api) {
 
+  // TODO(lambdai): Implement internal listener.
+  if (config.address().has_envoy_internal_address()) {
+    ENVOY_LOG(debug,
+              "listener {} has envoy internal address {}. Internal address cannot be used by "
+              "listener yet",
+              config.name(), config.address().envoy_internal_address().DebugString());
+    return false;
```

Comment:
Just use NOT_IMPLEMENTED? Or just ASSERT not has_envoy_internal_address? Seems not worth it to handle this case right now.

Reply:
What is the point of this PR? It's not that many lines of code and it doesn't actually do anything? I would rather you either a) just hold this PR and implement the full thing or b) put in ASSERT or RELEASE_ASSERT for stuff like this since you are have a not implemented and not documented feature. It doesn't make any sense to add this type of logic.

Reply:
I'd like to push api first as the conversion.

I don't get your plan b RELEASE_ASSERT is easier but it will crash Envoy. Why do you prefer an unsupported listener config crashing Envoy? With these lines above envoy would NACK the listener config without crash.

Reply:
Because you are adding error handling/logging logic for something that is not implemented. It bloats the code base for no good reason. If you want to make sure it crashes, do RELEASE_ASSERT(..., "this is not implemented yet."). Thanks.

/wait

Reply:
Oh, to NACK I should throw exception instead of return false.

Reply:
@mattklein123 Is throwing EnvoyException beating `return false` and RELEASE_ASSERT?

Reply:
No. Crash, or please just close this PR and reopen when there is an implementation. It does not make any sense to add actual error handling for an unimplemented partial feature.

Reply:
Happy to crash and move to next listener config api.
And then the internal connection driven by scheduler.

Running tests...